src/rime/algo/syllabifier.cc

```diff
@@ -74,20 +98,30 @@ int Syllabifier::BuildSyllableGraph(const string &input,
             props.end_pos = end_pos;
             // add a syllable with properties to the edge's
             // spelling-to-syllable map
-            spellings.insert({syllable_id, props});
+            if (matches_set.find(m.value) == matches_set.end()) {
+              props.is_correction = true;
+              props.credibility = 0.01;
+            }
+            auto it = spellings.find(syllable_id);
+            if (it == spellings.end()) {
+              spellings.insert({syllable_id, props});
+            } else {
+              it->second.type = std::min(it->second.type, props.type);
+            }
             // let end_vertex_type be the best (smaller) type of spelling
             // that ends at the vertex
-            if (end_vertex_type > props.type) {
+            if (end_vertex_type > props.type && !props.is_correction) {
```

Comment:
What if a position can only be reached via correction?

Reply:
It should be kept as normal spelling.

Reply:
Oh I see. End vertex of a correction edge should be marked as normal spelling even there is a worse typed edge overlapped.

Reply:
And the worse overlapped edge should be deleted?

Reply:
OK. It should keep the behavior as there isn't a correction.